ibm_mq/datadog_checks/ibm_mq/ibm_mq.py

```diff
@@ -220,16 +220,18 @@ def get_pcf_channel_metrics(self, queue_manager, tags, config):
             mname = '{}.channel.channels'.format(self.METRIC_PREFIX)
             self.gauge(mname, channels, tags=tags)
 
-        # grab all the discoverable channels
-        self._submit_channel_status(queue_manager, '*', tags, config)
-
-        # check specific channels as well
-        # if a channel is not listed in the above one, a user may want to check it specifically,
-        # in this case it'll fail
+        submitted_channels = set()  # record submitted channels to avoid duplicate submission
+        # Check specific channels
+        # If a channel is not discoverable, a user may want to check it specifically.
+        # Specific channels are checked first to send channel metrics and `ibm_mq.channel` service checks
+        # at the same time, but the end result is the same in any order.
         for channel in config.channels:
-            self._submit_channel_status(queue_manager, channel, tags, config)
+            self._submit_channel_status(queue_manager, channel, tags, config, submitted_channels)
```

Comment:
I don't like passing it submitted_channel to be modified by the `_submit_channel_status` method (it's a non obvious side effect of calling `_submit_channel_status`). I think it would be clearer if `_submit_channel_status` returned the list of submitted channels and explicitly adding them `submitted_channels`, that way also `submitted_channels` is responsibility of only one method and not two.

Reply:
Simplified the code to avoid having to store `submitted_channels`. Let me know what you think about it.